src/Content/Widget.php

```diff
@@ -374,81 +374,59 @@ public static function categories($baseurl, $selected = '')
 	}
 
 	/**
-	 * Return common friends visitor widget
+	 * Show a random selection of five common contacts between the visitor and the viewed profile user.
 	 *
-	 * @param string $profile_uid uid
+	 * @param int    $uid      Viewed profile user ID
+	 * @param string $nickname Viewed profile user nickname
 	 * @return string|void
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 * @throws \ImagickException
 	 */
-	public static function commonFriendsVisitor($profile_uid)
+	public static function commonFriendsVisitor(int $uid, string $nickname)
 	{
-		if (local_user() == $profile_uid) {
-			return;
-		}
-
-		$zcid = 0;
-
-		$cid = Session::getRemoteContactID($profile_uid);
-
-		if (!$cid) {
-			if (Profile::getMyURL()) {
-				$contact = DBA::selectFirst('contact', ['id'],
-						['nurl' => Strings::normaliseLink(Profile::getMyURL()), 'uid' => $profile_uid]);
-				if (DBA::isResult($contact)) {
-					$cid = $contact['id'];
-				} else {
-					$gcontact = DBA::selectFirst('gcontact', ['id'], ['nurl' => Strings::normaliseLink(Profile::getMyURL())]);
-					if (DBA::isResult($gcontact)) {
-						$zcid = $gcontact['id'];
-					}
-				}
-			}
+		if (local_user() == $uid) {
+			return '';
 		}
 
-		if ($cid == 0 && $zcid == 0) {
-			return;
+		$visitorPCid = local_user() ? Contact::getPublicIdByUserId(local_user()) : remote_user();
```

Comment:
Why aren't you using `Profile::getMyURL()` and `Session::getRemoteContactI` anymore?

Reply:
Because I've seen `Profile::getMyURL()` being populated even though `Session::isAuthenticated()` returned false and neither `local_user()` nor `remote_user()` returned an existing id, so I don't trust it for authentication. And we don't need it thanks of `remote_user()`.

For `Session::getRemoteContactId`, it returns a user contact id (uid != 0) which I would have to extract the public contact id, when it is already available in `remote_user()` from what I understand since the documentation is lacking for these functions/methods. Besides, you don't need to be in contact with a profile (which `Session::getRemoteContactId` seems to provide) to access the common contacts, you just need to be authenticated in general.

Reply:
AFAIK the function `Profile::getMyURL()` already returns a value when we connect with the other side with the `zrl=...` parameter. So the question is how to treat this.

Reply:
See https://github.com/friendica/friendica/blob/73c112066d309bd137cdba7adffd495290ebab91/src/App.php#L458-L478

`my_url` can be set while `authenticated` isn't. I don't know the exact conditions for this to happen but it looks like you just need to append `?zrl=<any profile url>` ? In any case I don't trust it for authentication, I only use `Profile::getMyURL()` once I ensured authentication.

Reply:
That's what I said above :-)

Reply:
Great, do you need more information or can this PR be merged as is?

Reply:
The morning I only had checked parts of the code. The evening I will check the rest.